### ISSUE TITLE
Super slow training on Voxceleb training

### ISSUE THREAD

**Issue by sun-peach** [hsn-zeinali/x-vector-kaldi-tf#2]
Hi, I am using the 2080 ti and tensorflow 1.13 (ubuntu 16.04, cuda 10.0, cudnn 7.0, python 3). It seems very slow to make progress in training on voxceleb  corpus (It has been 24 hours, but it still shows iter 0/167 without any progress). The following is the current log:

local/tf/train_dnn.py --stage=-1 --tf-model-class=ModelWithoutDropoutTdnn --cmd=run.pl --gpu 1 --num-targets=7323 --proportional-shrink=10 --minibatch-size=64 --max-param-change=2 --momentum=0.5 --num-jobs-initial=1 --num-jobs-final=1 --initial-effective-lrate=0.001 --final-effective-lrate=0.0001 --random-seed=2468 --num-epochs=2 --dropout-schedule=0,0@0.20,0.1@0.50,0 --egs-dir=/media/feit/Work/Work/SpeakerID/Kaldi_Voxceleb/exp/xvector_tf_but_test/egs --preserve-model-interval=10 --use-gpu=yes --dir=/media/feit/Work/Work/SpeakerID/Kaldi_Voxceleb/exp/xvector_tf_but_test
2019-05-22 18:44:33,764 [local/tf/train_dnn.py:473 - train - INFO ] Arguments for the experiment
{'cleanup': True,
 'command': 'run.pl --gpu 1',
 'dir': '/media/feit/Work/Work/SpeakerID/Kaldi_Voxceleb/exp/xvector_tf_but_test',
 'do_final_combination': False,
 'dropout_schedule': '0,0@0.20,0.1@0.50,0',
 'egs_dir': '/media/feit/Work/Work/SpeakerID/Kaldi_Voxceleb/exp/xvector_tf_but_test/egs',
 'final_effective_lrate': 0.0001,
 'initial_effective_lrate': 0.001,
 'max_models_combine': 20,
 'max_objective_evaluations': 30,
 'max_param_change': 2.0,
 'minibatch_size': 64,
 'momentum': 0.5,
 'num_epochs': 2.0,
 'num_jobs_final': 1,
 'num_jobs_initial': 1,
 'num_targets': 7323,
 'preserve_model_interval': 10,
 'proportional_shrink': 10.0,
 'random_seed': 2468,
 'stage': -1,
 'targets_scp': None,
 'tf_model_class': 'ModelWithoutDropoutTdnn',
 'use_gpu': 'yes'}
2019-05-22 18:44:33,765 [local/tf/train_dnn.py:489 - train - INFO ] Preparing the initial network.
2019-05-22 18:44:33,765 [local/tf/train_dnn.py:493 - train - INFO ] Start calling build_model to initialize the model ModelWithoutDropoutTdnn ...
2019-05-22 18:44:33,765 [/home/feit/Tool/x-vector-kaldi-tf/local/tf/models.py:551 - build_model - INFO ] Start building the model ...
WARNING:tensorflow:From /home/feit/Tool/anaconda3/envs/feitpy3base/lib/python3.5/site-packages/tensorflow/python/framework/op_def_library.py:263: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /home/feit/Tool/x-vector-kaldi-tf/local/tf/models.py:617: softmax_cross_entropy_with_logits (from tensorflow.python.ops.nn_ops) is deprecated and will be removed in a future version.
Instructions for updating:

Future major versions of TensorFlow will allow gradients to flow
into the labels input on backprop by default.

See `tf.nn.softmax_cross_entropy_with_logits_v2`.

WARNING:tensorflow:From /home/feit/Tool/anaconda3/envs/feitpy3base/lib/python3.5/site-packages/tensorflow/python/ops/math_ops.py:3066: to_int32 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
2019-05-22 18:44:35.897687: E tensorflow/stream_executor/cuda/cuda_driver.cc:300] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected

WARNING: The TensorFlow contrib module will not be included in TensorFlow 2.0.
For more information, please see:
  * https://github.com/tensorflow/community/blob/master/rfcs/20180907-contrib-sunset.md
  * https://github.com/tensorflow/addons
If you depend on functionality not listed there, please file an issue.

2019-05-22 18:44:35,902 [/home/feit/Tool/x-vector-kaldi-tf/local/tf/models.py:634 - build_model - INFO ] Start initializing the graph ...
2019-05-22 18:44:36,371 [/home/feit/Tool/x-vector-kaldi-tf/local/tf/models.py:133 - save_model - INFO ] Start saving graph ...
2019-05-22 18:44:36,803 [/home/feit/Tool/x-vector-kaldi-tf/local/tf/models.py:141 - save_model - INFO ] Graph saved in path: /media/feit/Work/Work/SpeakerID/Kaldi_Voxceleb/exp/xvector_tf_but_test/model_0/model
2019-05-22 18:44:36,813 [/home/feit/Tool/x-vector-kaldi-tf/local/tf/models.py:639 - build_model - INFO ] Building finished.
2019-05-22 18:44:36,813 [local/tf/train_dnn.py:516 - train - INFO ] Training will run for 2.0 epochs = 168 iterations
2019-05-22 18:44:36,813 [local/tf/train_dnn.py:544 - train - INFO ] Iter: 0/167    Epoch: 0.00/2.0 (0.0% complete)    lr: 0.001000    shrink: 0.99000

BTW, I change the code a little bit as below:
def set_cuda_visible_devices(use_gpu=True, logger=None):
    try:
        os.environ["CUDA_VISIBLE_DEVICES"] = '0'
        # if use_gpu:
        #     free_gpu = subprocess.check_output(cuda_command2, shell=True)
        #     if len(free_gpu) == 0:
        #         create_log_on_gpu_error()
        #         if logger is not None:
        #             logger.info("No GPU seems to be available and I cannot continue without GPU.")
        #         raise Exception("No GPU seems to be available and I cannot continue without GPU.")
        #     else:
        #         os.environ["CUDA_VISIBLE_DEVICES"] = free_gpu.decode().strip()
        #     if logger is not None:
        #         logger.info("CUDA_VISIBLE_DEVICES " + os.environ["CUDA_VISIBLE_DEVICES"])
        # else:
        #     os.environ["CUDA_VISIBLE_DEVICES"] = ''
        if not use_gpu:
            os.environ["CUDA_VISIBLE_DEVICES"] = ''
    except subprocess.CalledProcessError:
        if logger is not None:
            create_log_on_gpu_error()
            logger.info("No GPU seems to be available and I cannot continue without GPU.")
        # os.environ["CUDA_VISIBLE_DEVICES"] = ''
        if use_gpu:
            raise

Since I always have the issue when I use the original code. It seems the cuda_command2 give me empty string.

So is there anything wrong I did since it is very slow?

Thank you
